test/e2e/specs/meta-boxes.test.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Internal dependencies
+ */
+import '../support/bootstrap';
+import { newPost, newDesktopBrowserPage } from '../support/utils';
+import { activatePlugin, deactivatePlugin } from '../support/plugins';
+
+describe( 'Meta boxes', () => {
+	beforeAll( async () => {
+		await newDesktopBrowserPage();
+		await activatePlugin( 'gutenberg-test-plugin-meta-box' );
+		await newPost();
+	} );
+
+	afterAll( async () => {
+		await newDesktopBrowserPage();
+		await deactivatePlugin( 'gutenberg-test-plugin-meta-box' );
+	} );
+
+	it( 'Should save the post', async () => {
+		// Save should not be an option for new empty post.
+		expect( await page.$( '.editor-post-save-draft' ) ).toBe( null );
+
+		// Add title to enable valid non-empty post save.
+		await page.type( '.editor-post-title__input', 'Hello Meta' );
+		expect( await page.$( '.editor-post-save-draft' ) ).not.toBe( null );
+
+		await Promise.all( [
+			// Transitions between two states "Saving..." -> "Save Draft" (the
+			// button is always visible while meta are present).
```

Comment:
The button is hidden for 1 second though, we could also check the "saved" state between the two states :)

Reply:
>The button is hidden for 1 second though, we could also check the "saved" state between the two states :)

Ah, I missed this change come through. Added a test for it in 0c4d522, refactoring the PostSavedState a bit to apply modifier classes.